quickwit/quickwit-search/src/root.rs

```diff
@@ -394,6 +396,7 @@ pub async fn root_search(
 pub fn finalize_aggregation(
     intermediate_aggregation_result: Option<Vec<u8>>,
     aggregations: Option<QuickwitAggregations>,
+    searcher_context: &Arc<SearcherContext>,
```

Comment:
nit: Can you attempt to change the signature of `aggregation_limits_from_searcher_context ` and this function to `searcher_context: &SearcherContext`. If we're not cloning the `Arc` anywhere, it would be cleaner.

`SearcherContext` could also have an `aggregation_limits` accessor.